ROCm 6.3.0 libraries

AngryLoki opened the pull request:
This is a full set of ROCm libraries for currently existing packages.

It drops previously masked 6.2.x ebuilds, as they were incomplete (both from dependencies and quality checkpoint).
Tech notes, apart from https://rocm.docs.amd.com/en/latest/about/release-notes.html:
* New release migrated to LLVM 19. Technically, as usual, AMD team used some in-between release before llvm-19, so I added patches from upstream to fix compilation with official LLVM release. 
* No new GPUs added in this PR; 6.3.0 mentions new hardware, but it was not released yet, so changes are pretty much guaranteed.
* Apart from many bugfixes (as usual) subjectively the nicest one is https://github.com/llvm/llvm-project/issues/78344. Vanilla clang now outputs `Found HIP installation: /usr, version 6.3.42131` in `clang -v -print-targets`, which finally allows to write `enable_language(HIP)` in CMake and it will work.
* No updates for rocprofiler, as it is still stuck with proprietary dependency. But there are good news in https://github.com/ROCm/ROCm/issues/1781 - probably source will be open soon.

Closes: https://bugs.gentoo.org/939920
Closes: https://bugs.gentoo.org/940510
Closes: https://bugs.gentoo.org/937528
Closes: https://bugs.gentoo.org/937428
Closes: https://bugs.gentoo.org/935627
Closes: https://bugs.gentoo.org/945508
Closes: https://bugs.gentoo.org/935628
Closes: https://bugs.gentoo.org/945875
Closes: https://bugs.gentoo.org/944820

---

Please check all the boxes that apply:

- [x] I can submit this contribution in agreement with the [Copyright Policy](https://www.gentoo.org/glep/glep-0076.html#certificate-of-origin).
- [x] I have certified the above via adding a `Signed-off-by` line to *every* commit in the pull request.
- [x] This contribution has not been created with the assistance of Natural Language Processing artificial intelligence tools, in accordance with the [AI policy](https://wiki.gentoo.org/wiki/Project:Council/AI_policy).
- [x] I have run `pkgcheck scan --commits --net` to check for issues with my commits.

Please note that all boxes must be checked for the pull request to be merged.

AngryLoki committed:
sci-libs/hipBLAS-common: new package, add 6.3.0

Signed-off-by: Sv. Lockal <lockalsash@gmail.com>

AngryLoki committed:
dev-build/rocm-cmake: add 6.3.0

Signed-off-by: Sv. Lockal <lockalsash@gmail.com>

AngryLoki committed:
dev-libs/rocm-comgr: add 6.3.0, drop 6.2.4

Signed-off-by: Sv. Lockal <lockalsash@gmail.com>

AngryLoki committed:
dev-libs/rocm-device-libs: add 6.3.0, drop 6.2.4

Signed-off-by: Sv. Lockal <lockalsash@gmail.com>

AngryLoki committed:
dev-libs/rocm-opencl-runtime: add 6.3.0, drop 6.2.4

Apply must patch to 6.1.2 to fix compilation there too.

Closes: https://bugs.gentoo.org/939920
Signed-off-by: Sv. Lockal <lockalsash@gmail.com>

AngryLoki committed:
dev-libs/rocr-runtime: add 6.3.0, drop 6.2.4

Closes: https://bugs.gentoo.org/940510
Closes: https://bugs.gentoo.org/937528
Closes: https://bugs.gentoo.org/937428
Closes: https://bugs.gentoo.org/935627
Signed-off-by: Sv. Lockal <lockalsash@gmail.com>

AngryLoki committed:
dev-libs/roct-thunk-interface: add 6.3.0, drop 6.2.4

Also pin llvm version for 6.1.2, as it is not compatible with new LLVM.

Closes: https://bugs.gentoo.org/945508
Closes: https://bugs.gentoo.org/935628
Signed-off-by: Sv. Lockal <lockalsash@gmail.com>

AngryLoki committed:
dev-libs/hipother: add 6.3.0

Signed-off-by: Sv. Lockal <lockalsash@gmail.com>

AngryLoki committed:
dev-util/hip: add 6.3.0, drop 6.2.4

Closes: https://bugs.gentoo.org/945875
Signed-off-by: Sv. Lockal <lockalsash@gmail.com>

AngryLoki committed:
dev-util/hipcc: add 6.3.0, drop 6.2.4

Signed-off-by: Sv. Lockal <lockalsash@gmail.com>

AngryLoki committed:
dev-util/hipify-clang: add 6.3.0

Signed-off-by: Sv. Lockal <lockalsash@gmail.com>

AngryLoki committed:
dev-util/rocm-smi: add 6.3.0, drop 9999

Closes: https://bugs.gentoo.org/883409
Signed-off-by: Sv. Lockal <lockalsash@gmail.com>

AngryLoki committed:
dev-util/rocminfo: add 6.3.0

Signed-off-by: Sv. Lockal <lockalsash@gmail.com>

AngryLoki committed:
dev-util/Tensile: add 6.3.0

Signed-off-by: Sv. Lockal <lockalsash@gmail.com>

AngryLoki committed:
sci-libs/hipBLAS: add 6.3.0

Signed-off-by: Sv. Lockal <lockalsash@gmail.com>

AngryLoki committed:
sci-libs/hipBLASLt: add 6.3.0

Signed-off-by: Sv. Lockal <lockalsash@gmail.com>

AngryLoki committed:
sci-libs/hipSPARSE: add 6.3.0

Signed-off-by: Sv. Lockal <lockalsash@gmail.com>

AngryLoki committed:
sci-libs/rocSOLVER: add 6.3.0

Signed-off-by: Sv. Lockal <lockalsash@gmail.com>

AngryLoki committed:
sci-libs/rocPRIM: add 6.3.0

Signed-off-by: Sv. Lockal <lockalsash@gmail.com>

AngryLoki committed:
sci-libs/rocSPARSE: add 6.3.0

Signed-off-by: Sv. Lockal <lockalsash@gmail.com>

AngryLoki committed:
sci-libs/rocBLAS: add 6.3.0

Also backport fix for 16-bit floats for 6.1.1

Closes: https://bugs.gentoo.org/944820
Signed-off-by: Sv. Lockal <lockalsash@gmail.com>

AngryLoki committed:
dev-util/roctracer: add 6.3.0

Signed-off-by: Sv. Lockal <lockalsash@gmail.com>

AngryLoki committed:
sci-libs/composable-kernel: add 6.3.0

Signed-off-by: Sv. Lockal <lockalsash@gmail.com>

AngryLoki committed:
sci-libs/miopen: add 6.3.0

Signed-off-by: Sv. Lockal <lockalsash@gmail.com>

AngryLoki committed:
sci-libs/rocRAND: add 6.3.0

Signed-off-by: Sv. Lockal <lockalsash@gmail.com>

AngryLoki committed:
sci-libs/hipCUB: add 6.3.0

Signed-off-by: Sv. Lockal <lockalsash@gmail.com>

AngryLoki committed:
sci-libs/hipFFT: add 6.3.0

Signed-off-by: Sv. Lockal <lockalsash@gmail.com>

AngryLoki committed:
sci-libs/hipSOLVER: add 6.3.0

Signed-off-by: Sv. Lockal <lockalsash@gmail.com>

AngryLoki committed:
sci-libs/rocFFT: add 6.3.0

Signed-off-by: Sv. Lockal <lockalsash@gmail.com>

AngryLoki committed:
sci-libs/hipRAND: add 6.3.0

Signed-off-by: Sv. Lockal <lockalsash@gmail.com>

AngryLoki committed:
dev-libs/rocdbgapi: add 6.3.0

Signed-off-by: Sv. Lockal <lockalsash@gmail.com>

AngryLoki committed:
dev-util/rocm_bandwidth_test: add 6.3.0

Signed-off-by: Sv. Lockal <lockalsash@gmail.com>

AngryLoki committed:
sci-libs/rocWMMA: add 6.3.0

Signed-off-by: Sv. Lockal <lockalsash@gmail.com>

AngryLoki committed:
sci-libs/rocThrust: add 6.3.0

Signed-off-by: Sv. Lockal <lockalsash@gmail.com>

AngryLoki committed:
dev-libs/rccl: add 6.3.0

Signed-off-by: Sv. Lockal <lockalsash@gmail.com>

AngryLoki committed:
profiles: unmask ROCm 6.2 (ebuilds dropped in favor of 6.3)

Signed-off-by: Sv. Lockal <lockalsash@gmail.com>

AngryLoki commented:
@Patrick Lauer, @littlewu2508, @candrews et al, feel free to check this PR if you have time. I think it is complete, but it will also take me at least few days to build it with different use flags and compilers.

yretenai commented:
hipBLASLt supports more GFX targets:
https://github.com/ROCm/hipBLASLt/commit/938900a10d0c433eb8b90c2a3b65ba70ed39a91b
https://github.com/ROCm/hipBLASLt/commit/6b942aa35ddc7e410fb5594ac925ee7824c5d90c
https://github.com/ROCm/hipBLASLt/commit/19da2677cff3f93483600c2e39c79fad876422e4

full GPU list is now approximately `gfx908 gfx908 gfx90a gfx940 gfx941 gfx942 gfx1100 gfx1101 gfx1200 gfx1201` (probably more)

yretenai commented:
rocBLAS fails to emerge due to link-time error caused by missing `cblas_*` symbols. i'd upload the build log but it's 300 MB.

AngryLoki committed:
sci-libs/rpp: add 6.3.0

Signed-off-by: Sv. Lockal <lockalsash@gmail.com>

AngryLoki commented:
@yretenai , issue with rocBLAS fixed, also cleaned up warning flags to make logs smaller. 

Regarding hipBLASLt, I also added few targets, but without gfx12. The problem with these architectures is that hardware was not released yet, and adding them to "officially supported" list in `rocm.eclass` automatically enables associated USE flags, linearly increasing build space/time for those, who have not configured `AMD_TARGETS` (which might be a user problem, but again, this hardware was not even released yet, so no QA on that side).

yretenai commented:
Will test on my system tomorrow.

cmyster commented:
I am testing this as well. Please have a look at https://bpa.st/JBUQ
When upgrading from 6.1.2, we need to make sure dev-libs/rocm-comgr gets pulled before dev-libs/rocm-opencl-runtime.

Later it dies again here: https://bpa.st/BYYA

cmyster commented:
Tensile doesn't pass checksum verification:

!!! Fetched file: rocm-Tensile-6.3.0.tar.gz VERIFY FAILED!
!!! Reason: Filesize does not match recorded size
!!! Got:      12724004
!!! Expected: 12724464

cmyster commented:
sci-libs/hipBLAS needs sci-libs/hipBLAS-common built prior:

CMake Error at library/src/CMakeLists.txt:69 (find_package):
  Could not find a package configuration file provided by "hipblas-common"
  with any of the following names:

    hipblas-commonConfig.cmake
    hipblas-common-config.cmake

  Add the installation prefix of "hipblas-common" to CMAKE_PREFIX_PATH or set
  "hipblas-common_DIR" to a directory containing one of the above files.  If
  "hipblas-common" provides a separate development package or SDK, be sure it
  has been installed.

cmyster commented:
not sure if in scope of this change but please consider Caffe2 as well as it has this hardcodedness in all of its versions:
ROCM_VERSION=6.1

cmyster commented:
sci-libs/miopen fails to build: http://0x0.st/XFUN.log

cmyster commented:
sci-libs/hipSPARSE fails in configuration stage if dev-cpp/gtest is not installed: https://bpa.st/ORQA
But it later fails again at compilation stage http://0x0.st/XFGf.log

cmyster commented:
sci-libs/hipSPARSE won't compile: http://0x0.st/XFGC.log

cmyster commented:
sci-libs/rpp fails to compile: https://bpa.st/Y5HA

AngryLoki commented:
@cmyster , thanks, everything fixed and reported to upstream.

Regarding Caffe2 and other users of rocm libraries, I'd suggest not to mix them into this pull-request, as it is already too complex to review & merge by core team.

cmyster commented:
looks way better from compilation point of view: https://bpa.st/LJPQ
was tested with this --info https://bpa.st/R5NQ (I've set AMDGPU_TARGETS="gfx1100" manually)

cmyster commented:
dev-libs/rccl fails compilation phase http://0x0.st/XCgJ.log

cmyster commented:
dev-libs/rocm-comgr failed tests http://0x0.st/XCgd.log

voyageur commented:
I updated my installed packages with `AMDGPU_TARGETS: -* gfx1100`, most of the versions from this PR except:
dev-libs/hipother dev-libs/rccl
dev-util/hipify-clang dev-util/rocm_bandwidth_test
sci-libs/hipFFT sci-libs/hipSOLVER sci-libs/rocFFT sci-libs/rocWMMA sci-libs/rpp

Looking good from a quick test, except `sci-libs/miopen[composable-kernel]` failing to build (enabling other USE flags went OK): https://0x0.st/XCgl.log

Also, should these new USE-flags be enabled by default? composable-kernel hipblaslt rocblas
Upstream doc mentions them as prerequisites and enabled by default in CMakeLists.txt

cmyster commented:
dev-util/hip fails to fine math.h to compile its tests http://0x0.st/XCEB.log

cmyster commented:
dev-util/roctracer has a failing test  http://0x0.st/XCEx.log

cmyster commented:
sci-libs/hipCUB faild a test tests log: http://0x0.st/XC6K.log build log: http://0x0.st/XC6P.log

cmyster commented:
sci-libs/hipSPARSE fails at test phase: http://0x0.st/XC6Z.log

cmyster commented:
running the tests for sci-libs/miopen made my PC frozen to the point I had to force power off physically... http://0x0.st/XC6T.log

cmyster commented:
sci-libs/rocBLAS failed tests: http://0x0.st/XC6m.log

cmyster commented:
sci-libs/rocFFT can't find its tests: http://0x0.st/XC6j.log

cmyster commented:
sci-libs/rocPRIM faild to compile tests http://0x0.st/XC6e.log

cmyster commented:
sci-libs/rocSOLVER faild in test phase http://0x0.st/XC62.log

cmyster commented:
sci-libs/rocSPARSE failed in test phase http://0x0.st/XC6_.log

cmyster commented:
FYI when I talked to a maintainer about an issue with 6.2.4 I got this [reply](https://github.com/ROCm/composable_kernel/issues/1707#issuecomment-2513080602) saying that compilation needs to happen with AMD's LLVM. AFAIU this is not something that we do.

Bratzmeister commented:
> FYI when I talked to a maintainer about an issue with 6.2.4 I got this [reply](https://github.com/ROCm/composable_kernel/issues/1707#issuecomment-2513080602) saying that compilation needs to happen with AMD's LLVM. AFAIU this is not something that we do.

AMD also just released a version that's supposed to be used for rocm 6.3
https://github.com/ROCm/aomp/releases/tag/rel_20.0-1
>Updated to ROCm 6.3 sources

AngryLoki commented:
* dev-libs/rccl fixed by backporting 1 commit from develop, also disabled
* dev-util/hip math.h - fixed with a patch (workarounds around `-isystem /usr/include`)
* dev-libs/rocm-comgr failed tests: fixed in dev-util/hip
* dev-util/roctracer - one test disabled and reported
* sci-libs/hipBLASLt - added cblas to test dependencies, thanks
* sci-libs/hipCUB - maybe retry, also there should be `LastTest.log` file with failure details
* sci-libs/hipSPARSE and sci-libs/rocSPARSE - trip on the same problem with OpenMP, or kernel (kfd) issue
* sci-libs/miopen - I applied few changes to fix build with composable-kernel on gfx11*.
* sci-libs/rocBLAS - worth reporting to upstream, seems to be arch-related
* sci-libs/rocFFT - fixed path to test
* sci-libs/rocPRIM - found and backported fix from upstream
* sci-libs/rocSOLVER - "Queue 0x7f5c10200000 aborting" - I remember a similar issue with kernel (kfd)

> compilation needs to happen with AMD's LLVM

It is unlikely any distribution will ship AMD LLVM as long as they practice releases with "THIS RELEASE CANNOT BE BUILT FROM SOURCE EXTERNALLY" marks. Overall AMD uses internal development with approximately up to 6 months delay between "we pushed fix internally" and the final release (unless they backport the fix). Most of the changes related to codegen are upstreamed. To my knowledge there are no breaking issues that require using AMD LLVM specifically. As each release is attached to some prerelease version of LLVM and delay which I mentioned above, we see cases like 6.2.x and 6.3.0 which are not fully compatible neither with LLVM 18, nor LLVM 19. But with few backported patches from `amd-staging` branch it is usually possible to achieve this compatibility. 

At this moment it looks like AOMP is based on LLVM-20 prerelease, but as there are no stable API in nightly releases of [https://packages.gentoo.org/packages/llvm-core/clang](llvm-core/clang), I'd suggest to wait until release and not set `LLVM_COMPAT` until then (even though in theory it could fix some issues).

Regarding test completion, I highly suggest to upgrade to the latest kernel (which one you can). At some point in the past I even switched to `sys-kernel/git-sources` (for 6.10 kernel). I hope today situation is better, but just one year ago there were massive issues/improvements on kfd driver side (from memory leaks to deadlocks).

Regarding miopen and composable-kernel - official documentation describes how to build a result similar to official release. However, of you check out the source code, you will see that 90% of composable-kernel code xdl and mha kernels, enabled only for Instinct accelerators ([link](https://github.com/ROCm/composable_kernel/blob/07339c738396ebeae57374771ded4dcf11bddf1e/library/src/tensor_operation_instance/gpu/CMakeLists.txt#L50)). With gfx11* and gfx103* GPUs there are almost no code path even for tests (sure, on my gfx1030 `miopen[composable-kernel]` passes all tests, but it means nothing). And speaking of dependent software (like pytorch) even less chance to see that any of these kernels could be used. That's why I left a decision to enable these flags for user (or ebuild writers). If at any point somebody discovers a measurable improvement for any of these flags - sure, lets enable them by default.

Overall, @yretenai @voyageur @cmyster , thanks for testing. Overall priority is to have working rocr-runtime, hip/hipcc, roct-thunk-interface and rocm-comgr (used everywhere, fro blender/openimagedenoise to machine learning), smaller machine learning projects like llama.cpp and derivatives use just hipBLAS (which is already big), then pytorch uses almost everything and even more (mlir, triton, ...), I just hope that conditional enabling would help users to achieve their goals without requiring >64GB of ram and hours of compilation time.

LunNova commented:
roct-thunk-interface got made part of ROCR-Runtime upstream, is keeping it as a separate package here intentional?

AngryLoki commented:
> composable-kernel and rocThrust couldn't use my network

Fixed, thanks.

> Tensile and miopen both froze my machine a few tests in.

Restricted `test` feature there. It is unlikely that it fully passes the whole suite on non-instinct GPUs.

> roct-thunk-interface got made part of ROCR-Runtime upstream, is keeping it as a separate package here intentional?

Yes, it is intentional. ROCm developers are moving and merging repositories periodically. Overall, I think this move is not final anyways. As long as `hipcc` exists in `ROCm/llvm-project` and `amdgpu-arch` exists in `llvm/llvm-project`, `ROCm/{clr,HIP,rocminfo}` has a vector to move into llvm-project.

cmyster commented:
dev-libs/rocm-opencl-runtime has a test issue, maybe it should not even run:

./ocltst -m ./liboclgl.so -A ogl.exclude
Built for Emulation Environment

Compute Device info:
	Platform Version: OpenCL 2.1 AMD-APP.dbg (3635.0)
	Device Name: gfx1100
	Vendor: Advanced Micro Devices, Inc.
	Device Version: OpenCL 2.0
	Driver Version: 3635.0 (HSA1.1,LC)
	Board Name: AMD Radeon RX 7900 XTX
	Device Topology: PCI[ B#12, D#0, F#0]

-------------------------------------------------
The OpenCL Testing Module oclgl Version = 0
-------------------------------------------------
OCLGLBuffer[  0]                 PASSED: GPU[0] T[0] [  0],      0.000
OCLGLBufferMultipleQueues[  0]   PASSED: GPU[0] T[0] [  0],      0.000
OCLGLTexture[  0]                PASSED: GPU[0] T[0] [  0],      0.000
OCLGLTexture[  1]                PASSED: GPU[0] T[0] [  1],      0.000
OCLGLTexture[  2]                PASSED: GPU[0] T[0] [  2],      0.000
OCLGLTexture[  3]                PASSED: GPU[0] T[0] [  3],      0.000
OCLGLTexture[  4]                PASSED: GPU[0] T[0] [  4],      0.000
OCLGLTexture[  5]                PASSED: GPU[0] T[0] [  5],      0.000
OCLGLTexture[  6]                PASSED: GPU[0] T[0] [  6],      0.000
OCLGLTexture[  7]                PASSED: GPU[0] T[0] [  7],      0.000
OCLGLMultiContext[  0]           PASSED: GPU[0] T[0] [  0],      0.000
Silent failure: cl_khr_gl_event extension not available (-1)
OCLGLFenceSync[  0]              PASSED: GPU[0] T[0] [  0],      0.000
Silent failure: cl_khr_gl_event extension not available (-1)
OCLGLFenceSync[  1]              PASSED: GPU[0] T[0] [  1],      0.000
skipping test depth interop not supported
OCLGLDepthTex[  0]               PASSED: GPU[0] T[0] [  0],      0.000
skipping test depth interop not supported
OCLGLDepthTex[  1]               PASSED: GPU[0] T[0] [  1],      0.000
skipping test depth interop not supported
OCLGLDepthTex[  2]               PASSED: GPU[0] T[0] [  2],      0.000
skipping test depth interop not supported
OCLGLDepthTex[  3]               PASSED: GPU[0] T[0] [  3],      0.000
skipping test depth interop not supported
OCLGLDepthTex[  4]               PASSED: GPU[0] T[0] [  4],      0.000
skipping test depth interop not supported
OCLGLDepthTex[  5]               PASSED: GPU[0] T[0] [  5],      0.000
skipping test depth interop not supported
OCLGLDepthTex[  6]               PASSED: GPU[0] T[0] [  6],      0.000
skipping test depth interop not supported
OCLGLDepthTex[  7]               PASSED: GPU[0] T[0] [  7],      0.000
OCLGLPerfSepia[  0]             CL iterations# 50000 PASSED: GPU[0] T[0] [  0],      0.609
OCLGLPerfSepia[  1]             Segmentation fault

cmyster commented:
roct-thunk-interface has some tests failing - https://bpa.st/7UGA
Tensile with USE=test now wants me to install of these - https://bpa.st/UZPA
sci-libs/hipBLASLt LastTest.log has failures - http://0x0.st/8szH.log
sci-libs/hipCUB LastTest.log has failures - http://0x0.st/8szX.log
sci-libs/rocBLAS has test failures - http://0x0.st/8szP.log
sci-libs/rocPRIM LastTest.log has failures - http://0x0.st/8szZ.log

cmyster commented:
sci-libs/rocSOLVER ran several thousands of tests but finally failed on these https://pastebin.com/U78FwzQg
rocSPARSE started to run the tests but it never finished before I had an outage. I'll report the rest when I have results.

cmyster commented:
sci-libs/rocThrust LastTest.log has failures - http://0x0.st/8szv.log

cmyster commented:
finally, sci-libs/rocSPARSE throws some errors in its testing - http://0x0.st/8sXB.log

All the rest were fine, with either no tests execution or 100% pass rate.

AngryLoki commented:
@cmyster 

> Tensile with USE=test now wants me to install of these - https://bpa.st/UZPA

Don't use `USE=test emerge` or `FEATURES=test emerge` as it propagates. The easiest setup (or see [the wiki](https://wiki.gentoo.org/wiki/Package_testing)) is
```sh
echo 'FEATURES="test"' > /etc/portage/env/test
echo 'dev-util/Tensile test' > /etc/portage/package.env/test_packages
emerge -va1 dev-util/Tensile
```

Issues with `sci-libs/hipCUB` and `roct-thunk-interface` are gfx1100-related, I can't reproduce them. I see that these tests were already rewritten after release, so maybe the issue was fixed (but overall they seem not too important).

`sci-libs/hipBLASLt` test output looks 99% broken (first release with gfx1100 support btw). I'll try to check, even though I don't have gfx1100. It would be useful if you inspect/attach (after `emerge FEATURES="keeptemp" -va1 =sci-libs/hipBLASLt-6.3.0`) outputs of:

```
strace -e trace=open,openat -v /var/tmp/portage/sci-libs/hipBLASLt-6.3.0/work/hipBLASLt-rocm-6.3.0_build/clients/staging/hipblaslt-test
```

```
HIPBLASLT_LOG_LEVEL=5 AMD_LOG_LEVEL=7 AMD_COMGR_SAVE_TEMPS=1 AMD_COMGR_REDIRECT_LOGS=stdout AMD_COMGR_EMIT_VERBOSE_LOGS=1 /var/tmp/portage/sci-libs/hipBLASLt-6.3.0/work/hipBLASLt-rocm-6.3.0_build/clients/staging/hipblaslt-test
```

```
ltrace -C -e "hip*" -e "hsa*" /var/tmp/portage/sci-libs/hipBLASLt-6.3.0/work/hipBLASLt-rocm-6.3.0_build/clients/staging/hipblaslt-test
```

cmyster commented:
I think you are right @AngryLoki and the tests are not 'optimized' and my card is not well supported in those tests.

`strace -e trace=open,openat -v /var/tmp/portage/sci-libs/hipBLASLt-6.3.0/work/hipBLASLt-rocm-6.3.0_build/clients/staging/hipblaslt-test` - http://0x0.st/8zo5.log

`HIPBLASLT_LOG_LEVEL=5 AMD_LOG_LEVEL=7 AMD_COMGR_SAVE_TEMPS=1 AMD_COMGR_REDIRECT_LOGS=stdout AMD_COMGR_EMIT_VERBOSE_LOGS=1 /var/tmp/portage/sci-libs/hipBLASLt-6.3.0/work/hipBLASLt-rocm-6.3.0_build/clients/staging/hipblaslt-test` - https://drive.google.com/file/d/1kOTi83yxrr35QQGLkcosiwjEYh5iChWI/view?usp=sharing (xz, 1.7gb)


ltrace is taking a lot of time, I'll upload when I have it

AngryLoki commented:
Rebased, with following changes:
* `dev-util/rocm-smi` now respects LDFLAGS (closes https://bugs.gentoo.org/883409)
* `sci-libs/hipBLASLt` - fix test dependencies
* `sci-libs/hipCUB` - disable HMM (one tests fails), disable BlockShuffle test that fails in parallel runs (now passes 100% on gfx1100)
* `sci-libs/rocBLAS` - reduce test set (full tests takes >1h, smoke test takes few seconds), now passes 100% on gfx1100
* `sci-libs/rocSOLVER` - filter out second GPU in tests, reduce test set (full tests takes >1h), now passes 100% on gfx1100
* `sci-libs/rocWMMA` - filter out second GPU in tests (fixes tests)
* `sci-libs/miopen` - fix libc++ patch

@cmyster , I actually decided to buy 7900XTX to check it. It is still the best consumer-grade GPU from AMD at this moment. The problem with slow tests is mostly due to missing filters (CI in official repositories skips most of the test, so ebuilds could use the same filters).

I fixes one major issue in miopen tests, but I left tests disabled as it seemingly runs forever with max GPU utilization (supposedly causing overheat).

For `sci-libs/hipBLASLt`, strangely enough, tests fully work for me, with kernels 6.6.62-gentoo-dist, 6.10.0 (custom), 6.12.6-gentoo-dist. Checked clang version `19.1.6+libcxx` and `19.1.4`. Maybe you pinned/masked some package, not enough information to understand the issue.

For rocSOLVER/rocWMMA I noticed that some tests fail on dGPU when iGPU is not disabled in the BIOS, which is [a well-known issue](https://rocm.docs.amd.com/projects/radeon/en/latest/docs/prerequisites.html#disable-igpu), so I added `HIP_VISIBLE_DEVICES=0`, which solves this issue. I did not notice that this issue affects `sci-libs/hipBLASLt`, but maybe I'm missing something and other hardware interferes with tests.

thesamesam commented:
Thank you for putting in so much work to this.

cmyster commented:
@AngryLoki I like your dedication! Now you have 2 cards for testing.
(Yes I know this is slight necro at this point)
I am re running my scripts to test it again, I don't mind the time it takes...
Reason is that I want to make sure no last minute regressions, but so far into the tests it seems okay at around 75% of them done.
I run with "~amd64" so I am always at latest with Linux-6.12.8 (custom .config) and llvm 19.1.6 / GCC 14.2.1_p20241221 ...  
Reason why I find these specific failures? good question, this is why I don't filter out tests ;)

AngryLoki commented:
@cmyster , I guess I found the issue of hipBLASLt tests - https://github.com/ROCm/hipBLASLt/blob/rocm-6.3.0/clients/gtest/CMakeLists.txt#L78 - they add `-mf16c` flag, which is a feature of AVX512-capable CPUs. Could you check whether your `cpuid2cpuflags` has `f16c`?

And this is a common issue of ROCm libraries - https://github.com/search?q=org%3AROCm+-mf16c+language%3ACMake&type=code&l=CMake - I'll do something with it.

cmyster commented:
Not near my desk but I would think my 5900x has it...

On Mon, 6 Jan 2025, 13:12 Sv. Lockal, ***@***.***> wrote:

> @cmyster <https://github.com/cmyster> , I guess I found the issue of
> hipBLASLt tests -
> https://github.com/ROCm/hipBLASLt/blob/rocm-6.3.0/clients/gtest/CMakeLists.txt#L78
> - they add -mf16c flag, which is a feature of AVX512-capable CPUs. Could
> you check whether your cpuid2cpuflags has f16c?
>
> And this is a common issue of ROCm libraries -
> https://github.com/search?q=org%3AROCm+-mf16c+language%3ACMake&type=code&l=CMake
> - I'll do something with it.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/gentoo/gentoo/pull/39654#issuecomment-2572891798>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AA7PFDNK2MXD46Q4XPVS7XL2JJQI3AVCNFSM6AAAAABTJY3XZGVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDKNZSHA4TCNZZHA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

eli-schwartz commented on sci-libs/composable-kernel/composable-kernel-6.3.0.ebuild:
```diff
@@ -0,0 +1,119 @@
+EAPI=8
+
+ROCM_VERSION=${PV}
+PYTHON_COMPAT=( python3_{10..13} python3_13t )
```
Noticed this while looking at something else... this adds python freethreading support in the very *first* addition of python, is that actually:
- intentional
- tested
- important

AngryLoki replied:
I don't really remember that (i. e. I checked it earlier, long before 6.3.0).
Overall composable-kernel has no native python libs, so I don't see why it should not work.
Is it important - I don't think so, I just followed the notice of pkgcheck, as all dependencies were python3_13t-compatible.
